wagn/lib/decko/engine.rb

```diff
@@ -12,27 +12,39 @@
 require 'diff/lcs'
 require 'diffy'
 
+require 'wagn'
 
 module Decko
+
   class Engine < ::Rails::Engine
 
     paths.add "app/controllers", :with => 'rails/controllers', :eager_load => true
     paths.add 'gem-assets',      :with => 'rails/assets'
     paths.add 'config/routes',   :with => 'rails/engine-routes.rb'
-    paths.add 'lib/tasks',       :with => 'lib/wagn/tasks', :glob => '**/*.rake'
-    
+    paths.add 'lib/tasks',       :with => "#{::Wagn.gem_root}/lib/wagn/tasks", :glob => '**/*.rake'
+    paths.add 'lib/wagn/config/initializers',
+              :with => File.join( Wagn.gem_root, 'lib/wagn/config/initializers' ), :glob => "**/*.rb"
+
+    initializer 'decko.engine.load_config_initializers',  :after => :load_config_initializers do
+      paths['lib/wagn/config/initializers'].existent.sort.each do |initializer|
+        load(initializer)
+      end
+    end
+
+    initializer 'engine.copy_configs', :before => 'decko.engine.load_config_initializers' do
+      #this code should all be in Wagn somewhere, and it is now, gem-wize
+      #Ideally railties would do this for us, and this is needed for both use cases
+      Engine.paths['request_log']   = Wagn.paths['request_log']
+      Engine.paths['log']           = Wagn.paths['log']
+      Engine.paths['lib/tasks']     = Wagn.paths['lib/tasks']
+      Engine.paths['config/routes'] = Wagn.paths['config/routes']
+    end
 
     initializer :connect_on_load do
       Cardio.cache == ::Rails.cache
       ActiveSupport.on_load(:active_record) do
-        if defined? Wagn
-          #this code should all be in Wagn somewhere, I suspect.
-          Engine.paths['request_log'] = Wagn.paths['request_log']
-          Engine.paths['log']         = Wagn.paths['log']
-        else
-          Cardio.card_config ::Rails.application.config
-        end
-        
+        Cardio.cache == ::Rails.cache
```

Comment:
No, one of them is the right one, I'll delete the other.  And don't worry, I'll retest all the major scenarios once it is all stable.